cilium-cli/connectivity: allow to specify log levels to check

tklauser opened the pull request:
Since commit 5417a3fa37bc ("cli/connectivity: Check for unexpected warning logs") we check for unexpected warnings in logs as part of the connectivity checks. This leads to some fallout in cilium/cilium-cli when pulling in that change[^1] for the upcoming cilium-cli v0.16.21 release, mainly due to the fact that Hubble and Hubble UI are enabled in the cilium/cilium-cli workflows.

To work around that and don't block the next release of cilium-cli for too long, allow to specify log levels to check by means of a hidden flag (by default, errors and warnings are both still checked). This allows to skip checking of warnings logs temporarily. In the long term all these warning logs should be addressed by either fixing their root cause, demoting their level or adding them to the ignore list.

[^1]: https://github.com/cilium/cilium-cli/pull/2869

/cc @pchaigno

mhofstetter commented on cilium-cli/cli/connectivity.go:
```diff
@@ -178,6 +178,9 @@ func newCmdConnectivityTest(hooks api.Hooks) *cobra.Command {
 	cmd.Flags().StringSliceVar(&params.ExpectedXFRMErrors, "expected-xfrm-errors", defaults.ExpectedXFRMErrors, "List of expected XFRM errors")
 	cmd.Flags().MarkHidden("expected-xfrm-errors")
 
+	cmd.Flags().StringSliceVar(&params.LogCheckLevels, "log-check-levels", defaults.LogCheckLevels, "Log levels to check for in error logs")
```
nit: `Log levels to check for in log messages` ? (as it's actually not an `error log`)

tklauser replied:
Good point, fixed.

mhofstetter reviewed (approved):
LGTM - thanks!

tklauser committed:
cilium-cli/connectivity: allow to specify log levels to check

Since commit 5417a3fa37bc ("cli/connectivity: Check for unexpected
warning logs") we check for unexpected warnings in logs as part of the
connectivity checks. This leads to some fallout in cilium/cilium-cli
when pulling in that change[^1] for the upcoming cilium-cli v0.16.21
release, mainly due to the fact that Hubble and Hubble UI are enabled in
the cilium/cilium-cli workflows.

To work around that and don't block the next release of cilium-cli for
too long, allow to specify log levels to check by means of a hidden
flag (by default, errors and warnings are both still checked). This
allows to skip checking of warnings logs temporarily. In the long term
all these warning logs should be addressed by either fixing their root
cause, demoting their level or adding them to the ignore list.

[^1]: https://github.com/cilium/cilium-cli/pull/2869

Signed-off-by: Tobias Klauser <tobias@cilium.io>
